src/main/java/uk/gov/ida/matchingserviceadapter/configuration/EuropeanIdentityConfiguration.java

```diff
@@ -30,10 +35,11 @@ public String getHubConnectorEntityId() {
         return hubConnectorEntityId;
     }
 
-    public String[] getAcceptableHubConnectorEntityIds() {
-        return acceptableHubConnectorEntityIds != null && acceptableHubConnectorEntityIds.length > 0
-            ? acceptableHubConnectorEntityIds
-            : new String[] { hubConnectorEntityId };
+    public List<String> getAcceptableHubConnectorEntityIds() {
+        Set<String> entityIds = new LinkedHashSet<>();
+        Optional.ofNullable(acceptableHubConnectorEntityIds).ifPresent(entityIds::addAll);
+        Optional.ofNullable(hubConnectorEntityId).ifPresent(entityIds::add);
+        return new LinkedList<>(entityIds);
```

Comment:
Could you explain this bit please?
From what I can see, you're trying to merge `acceptableHubConnectorEntityIds` and the current `hubConnectorEntityId` into an ordered set. Is the intention that when the `getAcceptableHubConnectorEntityIds` method is used, we evaluate the current `hubConnectorEntityId` last?

Reply:
This was overkill. It's enough to use a set, and the intention was to prevent duplication. The next exciting update shifts down to a regular HashSet and does some other fun things with default values 👍